.github/workflows/release.yml

```diff
@@ -0,0 +1,25 @@
+---
+name: Release openqasm3_parser
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  publish_lexer:
+    name: Publish oq3_lexer crate
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        crate: [oq3_lexer, oq3_parser, oq3_semantics, oq3_sourcegen, oq3_syntax, oq3_source_file]
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - name: Run cargo publish
+        run: |
+          cd crates/${{ matrix.crate }}
+          cargo publish
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
```

Comment:
Do the crates get verified crates.io-side at all?  If so, we might have to force the order of them to some topological iteration through them to get them uploaded.

Seems a shame that there's no workspace-wide `cargo publish`.

Reply:
Oh that's a good point, yeah we'll need to do this manually in topological order otherwise cargo's validation will error when the dependencies aren't resolvable with crates.io. You get an error like:

```
error: failed to prepare local package for uploading

Caused by:
  no matching package named `oq3_lexer` found
  location searched: registry `crates-io`
```

Reply:
If this is true https://github.com/rust-lang/cargo/issues/1169#issuecomment-1756319495 , then there is no problem publishing out of order, as long as the crates have been published once in the correct order (say manually).

Reply:
There is a ton of discussion around this and many tools available. This tool looks to me (fwiw) to be the most serious and perhaps relevant.
https://github.com/crate-ci/cargo-release